mockingbird/src/commonMain/kotlin/com/careem/mockingbird/test/Functions.kt

```diff
@@ -192,20 +199,25 @@ public fun <T : Spy, R> T.spy(
     arguments: Map<String, Any?> = emptyMap(),
     delegate: () -> R
 ): R {
-    val hashCode = this.hashCode()
+    val uuid = this.uuid
     return MockingBird.invocationRecorder().access { recorder ->
         val invocation = Invocation(methodName = methodName, arguments = arguments)
-        recordInvocation(hashCode, recorder, invocation)// TODO change name
+        recordInvocation(uuid, recorder, invocation)// TODO change name
         @Suppress("UNCHECKED_CAST")
         val mockResponse = recorder.getResponse(
-            instanceHash = hashCode,
+            uuid = uuid,
             invocation = invocation,
             relaxed = true
         ) as R
         return@access mockResponse ?: delegate()
     }
 }
 
+
+public fun <T : Mock> T.uuid(): Lazy<String> {
```

Comment:
utility functions for manual mock generation, same as `mock`, `mockUnit` and so on